sparsify/__main__.py

```diff
@@ -91,50 +92,50 @@ def load_artifacts(
             else None
         ),
         revision=args.revision,
-        torch_dtype=dtype,
+        dtype=dtype,  # ✅ replaces deprecated torch_dtype
         token=args.hf_token,
     )
 
-    # For memmap-style datasets
-    if args.dataset.endswith(".bin"):
-        dataset = MemmapDataset(args.dataset, args.ctx_len, args.max_examples)
-    else:
-        # For Huggingface datasets
-        try:
-            dataset = load_dataset(
-                args.dataset,
-                split=args.split,
-                # TODO: Maybe set this to False by default? But RPJ requires it.
-                trust_remote_code=True,
-            )
-        except ValueError as e:
-            # Automatically use load_from_disk if appropriate
-            if "load_from_disk" in str(e):
-                dataset = Dataset.load_from_disk(args.dataset, keep_in_memory=False)
-            else:
-                raise e
-
-        assert isinstance(dataset, Dataset)
-        if "input_ids" not in dataset.column_names:
+    def prepare_dataset(split: str) -> Dataset | MemmapDataset:
+        if args.dataset.endswith(".bin"):
+            return MemmapDataset(args.dataset, args.ctx_len, args.max_examples)
+
+        ds = load_dataset(
+            args.dataset,
+            split=split,
+            trust_remote_code=True,
+        )
+
+        assert isinstance(ds, Dataset)
+        if "input_ids" not in ds.column_names:
             tokenizer = AutoTokenizer.from_pretrained(args.model, token=args.hf_token)
-            dataset = chunk_and_tokenize(
-                dataset,
+            ds = chunk_and_tokenize(
+                ds,
                 tokenizer,
                 max_seq_len=args.ctx_len,
                 num_proc=args.data_preprocessing_num_proc,
                 text_key=args.text_column,
             )
         else:
-            print("Dataset already tokenized; skipping tokenization.")
+            print(f"Split '{split}' already tokenized; skipping tokenization.")
 
-        print(f"Shuffling dataset with seed {args.shuffle_seed}")
-        dataset = dataset.shuffle(args.shuffle_seed)
-
-        dataset = dataset.with_format("torch")
+        print(f"Shuffling split '{split}' with seed {args.shuffle_seed}")
+        ds = ds.shuffle(args.shuffle_seed).with_format("torch")
         if limit := args.max_examples:
-            dataset = dataset.select(range(limit))
+            ds = ds.select(range(limit))
+        return ds
+
+    # Train dataset
+    dataset = prepare_dataset(args.split)
+    print(f"Train dataset size: {len(dataset)}")
 
-    return model, dataset
+    # Optional eval dataset
+    eval_dataset = None
+    if args.eval_split:
+        eval_dataset = prepare_dataset(args.eval_split)
+        print(f"Eval dataset size: {len(eval_dataset)}")
+
+    return model, dataset, eval_dataset
```

Comment:
don't betray details of development history in naming conventions, i.e. instead of dataset (clearly first addition) and eval_dataset (clearly second addition) you want train_dataset and eval_dataset so they're on equal terms